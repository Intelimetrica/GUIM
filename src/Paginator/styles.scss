
@import '../styles/colors.scss';

.GUIMPaginator {
  width: 60%;
  padding: 0;
  list-style: none;
  display: flex;
  justify-content: space-around;
}
.GUIMPaginator li {
  cursor: pointer;
}
.GUIMPaginator li.Active {
  background-color: var( --main-blue);
}
.GUIMPaginator li.Disabled {
  opacity: 0.7;
}