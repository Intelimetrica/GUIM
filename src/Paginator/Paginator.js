



import React, { Component } from 'react';
import ReactPaginate from 'react-paginate';

const INITIAL_SELECTED_PAGE = 0;

import "./styles.scss";

class Paginator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      forcePage: undefined,
      meta: {
        limit: parseInt(props.meta.limit),
        offset: parseInt(props.meta.offset),
        total: parseInt(props.meta.total)
      },
      pageCount: Math.ceil(parseInt(props.meta.total) / parseInt(props.meta.limit))
    };
  }

  componentDidUpdate(prevProps) {
    // Super hack
    const { meta: { limit, offset, total } } = this.props;
    if (prevProps.meta.total !== total) {
      // If meta.total has changes the selected page should be updated
      let forcePage = INITIAL_SELECTED_PAGE;
      const limit_int = parseInt(limit);
      const offset_int = parseInt(offset);
      if (offset_int > 0) {
        forcePage = Math.floor(offset_int / limit_int);
      }
      this.setState({
        forcePage,
        meta: {
          limit: limit_int,
          offset: offset_int,
          total: parseInt(total)
        },
        pageCount: Math.ceil(parseInt(total) / parseInt(limit))
      });
    } else if (this.state.forcePage !== undefined) {
      // After the reset is done we go back to the default forcePage behavior
      this.setState({
        forcePage: undefined
      });
    }
  }

  render() {
    const { pages, pagesEachSideOfBreak, disabled } = this.props;
    const { forcePage, meta: { limit, offset, total }, pageCount } = this.state;
    return (
      <ReactPaginate
        previousLabel={`<`}
        nextLabel={`>`}
        breakLabel='...'
        breakClassName='PageDots'
        pageCount={pageCount}
        forcePage={forcePage}
        marginPagesDisplayed={pagesEachSideOfBreak}
        pageRangeDisplayed={pages}
        onPageChange={({ selected }) => this.props.onUpdate({
          offset: selected * limit,
          limit
        })}
        containerClassName={`GUIMPaginator ${disabled ? 'Disabled' : ''}`}
        activeClassName='Active'
        disabledClassName='Disabled'
        previousClassName='Previous'
        nextClassName='Next'
      />
    );
  }
}

Paginator.defaultProps = {
  meta: {
    limit: 10,
    offset: 0,
    total: 10
  },
  onUpdate: () => { },
  pages: 4,
  pagesEachSideOfBreak: 2,
  disabled: false
};

export default Paginator;