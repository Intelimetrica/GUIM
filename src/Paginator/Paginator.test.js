import React from 'react';
import Paginator from './Paginator';
import renderer from 'react-test-renderer';

  describe('<Paginator />', () => {
    describe('tests for normal behaviour', () => {
      let active = [1];
      let component;
      let tree;
      let reRender;
      const makeRenderer = component =>
        () => tree = component.toJSON();
  
      beforeEach(() => {
        component = renderer.create(<Paginator />);
        reRender = makeRenderer(component);
        reRender();
      })
  
      it('match snapshot', () => {
        expect(tree).toMatchSnapshot();
      });
    });
  });
  