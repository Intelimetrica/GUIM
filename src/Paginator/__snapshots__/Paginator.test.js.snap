// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Paginator /> tests for normal behaviour match snapshot 1`] = `
<ul
  className="GUIMPaginator "
>
  <li
    className="Previous Disabled"
  >
    <a
      aria-disabled="true"
      aria-label="Previous page"
      className={undefined}
      href={undefined}
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="prev"
      role="button"
      tabIndex="0"
    >
      &lt;
    </a>
  </li>
  <li
    className="Active"
  >
    <a
      aria-current="page"
      aria-label="Page 1 is your current page"
      className={undefined}
      href={undefined}
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex="0"
    >
      1
    </a>
  </li>
  <li
    className="Next Disabled"
  >
    <a
      aria-disabled="true"
      aria-label="Next page"
      className={undefined}
      href={undefined}
      onClick={[Function]}
      onKeyPress={[Function]}
      rel="next"
      role="button"
      tabIndex="0"
    >
      &gt;
    </a>
  </li>
</ul>
`;
